src/scales/scale.time.js

```diff
@@ -36,10 +40,94 @@ module.exports = function(Chart) {
 			},
 		},
 		ticks: {
-			autoSkip: false
+			autoSkip: false,
+			mode: 'linear',   // 'linear|series'
+			source: 'auto'    // 'auto|labels'
 		}
 	};
 
+	function buildLookupTable(ticks, min, max, linear) {
+		var ilen = ticks.length;
+		var table = [];
+		var i, prev, curr, next, decimal;
+
+		if (ilen === 0) {
+			return table;
+		}
+
+		if (ticks[0] !== min) {
+			table.push({time: min, decimal: 0});
+		}
+
+		for (i = 0, ilen=ticks.length; i<ilen; ++i) {
+			next = ticks[i + 1] || 0;
+			prev = ticks[i - 1] || 0;
+			curr = ticks[i];
+
+			// only add points that breaks the scale linearity
```

Comment:
can you expand upon this either here or in the main docs for the lookup table. It's not immediately clear to me what the structure of this table is or what this logic is for